commons/src/lib.rs

```diff
@@ -171,20 +208,83 @@ mod tests {
+        let accept_default = header::HeaderValue::from_str("application/json").unwrap();
+        let version = validate_content_type(
+            &headers,
+            vec![accept_default.clone()],
+            accept_default.clone(),
+        )
+        .unwrap(); // if the request leaves Accept empty, we return the minimum supported cincinnati version as that's the lowest version we support
+        assert_eq!(version, MIN_CINCINNATI_VERSION.to_string());
+
+        // `application/json` provided with header, server accepts `application/json` and defaults to `application/json`
         headers.insert(
             header::ACCEPT,
             //"application/json, text/*; q=0.2".parse().unwrap(), // prefer JSON, but also accept any text/* after an 80% markdown in quality.  FIXME: needs a smarter parser in validate_content_type
             "application/json".parse().unwrap(),
         );
-        validate_content_type(&headers, "application/json").unwrap();
+        let version = validate_content_type(
+            &headers,
+            vec![accept_default.clone()],
+            accept_default.clone(),
+        )
+        .unwrap();
+        assert_eq!(version, "application/json");
+
+        // `application/*` provided with header, server accepts `application/json` and defaults to `application/json`
+        headers.insert(header::ACCEPT, "application/*".parse().unwrap());
+        let version = validate_content_type(
+            &headers,
+            vec![accept_default.clone()],
+            accept_default.clone(),
+        )
+        .unwrap();
+        assert_eq!(version, "application/json");
+
+        // `image/png` provided with header, server accepts `application/json` and defaults to `application/json`
+        let image_type: Vec<actix_web::http::HeaderValue> =
+            vec![header::HeaderValue::from_str("image/png").unwrap()];
+        //server should throw error on non-supported ACCEPT
+        validate_content_type(&headers, image_type, accept_default.clone()).unwrap_err();
+
+        //`application/vnd.redhat.cincinnati.v1+json` provided with header, server accepts
+        //`application/vnd.redhat.cincinnati.v1+json` and defaults to `application/json`
+        headers.insert(
+            header::ACCEPT,
+            "application/vnd.redhat.cincinnati.v1+json".parse().unwrap(),
+        );
+        let cincinnati_v1: Vec<actix_web::http::HeaderValue> =
+            vec![
+                header::HeaderValue::from_str("application/vnd.redhat.cincinnati.v1+json").unwrap(),
+            ];
+        let version =
+            validate_content_type(&headers, cincinnati_v1.clone(), accept_default.clone()).unwrap();
+        // Server returns the response with content_type `application/vnd.redhat.cincinnati.v1+json`
+        assert_eq!(version, "application/vnd.redhat.cincinnati.v1+json");
```

Comment:
```suggestion
        assert_eq!(version, MIN_CINCINNATI_VERSION.to_string());
```

Reply:
updated the comments for tests to make it more clear and reduced hardcodes.